pandas/core/arrays/base.py

```diff
@@ -653,15 +661,59 @@ def copy(self, deep=False):
         raise AbstractMethodError(self)
 
     # ------------------------------------------------------------------------
-    # Block-related methods
+    # Printing
     # ------------------------------------------------------------------------
+    def __unicode__(self):
+        from pandas.io.formats.printing import format_object_summary
+
+        template = (
+            u'<{class_name}>\n'
```

Comment:
this is an abitrary repr here. This needs to justify why not using parens and not comma separated. -1 on having this different from the Index reprs that we already have.

Reply:
> this is an abitrary repr here. This needs to justify why not using parens and not comma separated.

Agreed. It's an arbitrary repr, and the justification is that we can't assume that it's valid code for the class inheriting it.

Reply:
@jreback see also the discussion in https://github.com/pandas-dev/pandas/issues/22846

Reply:
rather than creating a new arbitrary rep, let's just use what we have.